org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/IMaven.java

```diff
@@ -215,6 +216,14 @@ <T> T getMojoParameterValue(MavenProject project, String parameter, Class<T> typ
 
   List<SettingsProblem> validateSettings(String settings);
 
+  /**
+   * Validates the toolchains definition
+   * 
+   * @param toolchains The path to the toolchains definition file to test.
+   * @return List of all problems. Is never <code>null</code>.
+   */
+  List<Problem> validateToolchains(String toolchains);
```

Comment:
I think we should not add new methods to `IMaven` anymore, can you addit to the `org.eclipse.m2e.core.internal.IMavenToolbox` instead?

Reply:
Null problemo. method relocated.